### DISCUSSION TITLE
using ASSA graphics attachments

### DISCUSSION THREAD

**Discussion by byskyphy** [SubtitleEdit/subtitleedit#8702] [Q&A]
Hi everyone, newbie here with a niche question: I found the option to attach files, but for the life of me I can't figure out how to use them. I do see them in the source view, but I can't find how to apply them. 

![image](https://github.com/user-attachments/assets/c376cc8b-4325-421f-bf37-d7850e6ee00e)

The official FAQ is silent regarding this subject, unfortunately. After reading the .ass documentation [here](http://www.tcax.org/docs/ass-specs.htm) I assume that if was working with the source code, I'd add a separate "picture" line, but I still can't make it work inside SE nor with manual notepad edit.

I am also aware that the ttml format supports picture [embeds ](https://www.w3.org/TR/2018/REC-ttml2-20181108/#embedded-content-vocabulary-image) which is what I was initially interested in using. Not sure that's doable within SE as I haven't found any information on it yet. 

P.S. : My usecase is adding avatar icons to character's dialogue sentences, I'm trying to make motion graphics for a podcast.

**Comment by niksedk** [comment]
This is a good question. I've tried to show images too, but failed.

You can use the plug-in AssaDraw in SE to make something like this:
````
[Script Info]
; This is an Advanced Sub Station Alpha v4+ script.
Title: 
ScriptType: v4.00+
PlayDepth: 0
ScaledBorderAndShadow: Yes
PlayResX: 1280
PlayResY: 640
SelectedLines: 0

[V4+ Styles]
Format: Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, OutlineColour, BackColour, Bold, Italic, Underline, StrikeOut, ScaleX, ScaleY, Spacing, Angle, BorderStyle, Outline, Shadow, Alignment, MarginL, MarginR, MarginV, Encoding
Style: Default,Arial,44.4,&H00FFFFFF,&H0000FFFF,&H00000000,&H00000000,0,0,0,0,100,100,0,0,1,1,1,2,10,10,10,1
Style: AssaDraw,Arial,20,&H00FFFFFF,&H0000FFFF,&H00000000,&H00000000,0,0,0,0,100,100,0,0,1,0,0,7,0,0,0,1

[Events]
Format: Layer, Start, End, Style, Name, MarginL, MarginR, MarginV, Effect, Text
Dialogue: 1,0:00:01.00,0:00:03.00,AssaDraw,,0,0,0,,{\c&HFFFFFF&\p1}m 133 114 l 197 114 197 141 150 141 b 147 131 143 123 133 114  m 150 174 l 197 174 225 202 132 202 b 143 193 147 184 150 174  m 73 52 l 104 52 132 80 77 80 b 57 78 47 108 77 113 83 113 89 113 95 113 152 118 153 195 97 202 l 29 201 29 201 57 174 96 174 b 120 167 110 142 98 142 88 142 80 142 72 142 13 135 19 55 73 52  m 122 52 l 149 80 197 80 225 52{\p0}
Dialogue: 0,0:00:01.00,0:00:03.00,AssaDraw,,0,0,0,,{\c&H0000FF&\p1}m 2 216 l 1 45 b 0 16 26 -1 48 1 l 210 0 b 236 2 252 18 254 43 l 254 211 b 254 232 236 251 216 253 l 44 253 b 16 252 3 228 2 216{\p0}

````

![image](https://github.com/user-attachments/assets/65566990-a579-43a7-aa26-e1cbe9048120)


Also, @LeonCheung gave me this link, which can import some svg files: https://qgustavor.github.io/svg2ass-gui/

**Reply by byskyphy** [reply]
Hm... I will try this out, although adding more and more workarounds seems counterproductive. 

But what can I do with imported files right now ? Is there an explanation or a tutorial of the validated uses of the attachments feature ?

**Reply by niksedk** [reply]
It seems the picture stuff is unsupported: https://github.com/libass/libass/issues/253
I should probably remove the picture/image stuff from ASSA attachments in SE...

You can however use attached fonts :)

**Reply by niksedk** [reply]
Note: SE can also generate more advanced subtitle boxes (needs to be last step when subs are done):

![image](https://github.com/user-attachments/assets/666f293d-f02e-4c29-af6b-0e4e4010987b)

**Reply by byskyphy** [reply]
good to know, thank you. however, playing around with .ass isn't really what I'm ultimately pursuing. I'm trying to find an efficient workflow for transcribing & subtitling with AI to stylize them later on the web pages and/or in the videos. ass subs aren't great for that. I'm looking to work both with entire sentences and word-level time stamps.

by pure chance, I found [this project](https://github.com/hyperaudio/hyperaudio-lite-editor) today that does exactly that. their format seems quite more efficient and the web player works well, but the editor is incomplete.

**Comment by byskyphy** [comment]
On a related topic: Is there / will there be a transcription space that allows us to focus on sentences and speakers (actors) while maintaining timestamps in the source view?

Here's an example of this feature from Premiere Pro's autogenerated transcript 
![image](https://github.com/user-attachments/assets/badb31a9-8f40-471b-9470-06cca8bcede2)
